qlib/DebugProgramControl.qm

```
@@ -928,7 +1011,19 @@ public namespace DebugProgramControl {
                 ctx.intrInfo.file = location.file;
                 ctx.intrInfo.line = location.start_line;
                 ctx.intrInfo.endline = location.end_line;
-                #ctx.intrInfo.statement = location;
+                if (location.source != "") {
+                    ctx.intrInfo.source = location.source;
+                }
+                ctx.intrInfo.offset = location.offset;
+                try {
+                    ctx.intrInfo.linecode = getSourceCodeFromFileOrLabel(pgm, (
+                        'file': location.file,
+                        'line': location.start_line,
+                        'source': location.source != "" ? location.source : NOTHING,
+                        'offset': location.offset,
+                    ), 0, 0).code;
+                } catch (hash<ExceptionInfo> ex) {
```

are we intentionally suppressing exceptions? if so, please add comment to the catch block like "exception intentionally suppressed because ..."